### PR TITLE
Return null from render when no icon is found

### PR DESCRIPTION
**Pull request by mlwilkerson**
- Return null from render when no icon is found
- Add negative test case for not found icons
- Add listeners to logger to use in test

### CODE REVIEW

File: src/components/__tests__/FontAwesomeIcon.test.js [hunk 1]
```diff
@@ -54,6 +59,16 @@ test('using pack common names', () => {
...
 })
 
+test('using pack common names not added to library', () => {
+  const loggerCallback = jest.fn()
+  const loggerCallbackKey = `${Math.random()}-${Math.random()}-${Math.random()}`
+  listen(loggerCallbackKey, loggerCallback)
+  const vm = mount({ icon: 'spinner' })
+  expect(vm).toBeNull()
+  expect(loggerCallback.mock.calls.length).toBe(1)
+  expect(loggerCallback.mock.calls[0][0]).toEqual(expect.stringContaining('Could not find icon'))
```
Comment (robmadole) [inline src/components/__tests__/FontAwesomeIcon.test.js]: Nice!

File: src/components/FontAwesomeIcon.js [hunk 2]
```diff
@@ -60,7 +60,10 @@ function FontAwesomeIcon (props) {
     symbol
   })
 
-  if (!renderedIcon) return log('Could not find icon', icon)
+  if (!renderedIcon){
+    log('Could not find icon', icon)
+    return null
```
Comment (robmadole) [inline src/components/FontAwesomeIcon.js]: :+1:

File: src/logger.js [hunk 3]
```diff
@@ -1,11 +1,37 @@
...
+    listeners = {};
+  }
+}
+
 export default function (...args) {
-  if (!PRODUCTION && console && typeof console.error === 'function') {
+  if(TEST && Object.keys(listeners).length > 0) {
+    Object.keys(listeners).forEach(function(key){listeners[key](...args)});
+  }
+  else if (!PRODUCTION && console && typeof console.error === 'function') {
```
Comment (robmadole) [inline src/logger.js]: I wonder if we could accomplish this with Jest mocks.

https://facebook.github.io/jest/docs/en/manual-mocks.html#content

Related: https://stackoverflow.com/questions/41223963/jest-how-to-mock-console-when-it-is-used-by-a-third-party-library

Comment (mlwilkerson) [reply src/logger.js]: Yes, that's actually what I was looking for. But at first glance, it didn't seem like they were going to serve the purpose I had in mind. On a closer reading, yes, the mocks approach looks better. I'll tweak.

**Comment by robmadole** [comment]
awesome :+1: 🚢 it. Feel free to `Squash and merge` when you are ready @mlwilkerson

**Pull request merged**